app/src/pages/project/metrics/ProjectMetricsPage.tsx

```diff
@@ -129,6 +129,7 @@ function useClosedTimeRange(): EpochTimeRange {
 
   // Use a ref to freeze "now" until the context time range actually changes
   const lastTimestampsRef = useRef({ startMs, endMs });
+  // eslint-disable-next-line react-hooks/purity
```

Comment:
this hook already is using refs in the render loop. I'm not sure there will ever be a good solution for open time ranges other than this.